### PR TITLE
Improved semaphore error handling

### PR DESCRIPTION
**Pull request by vinhowe**
- [x] Test `_windows_check_process_alive` on a Windows machine
- [x] Print out log paths after an engine crash (will wait for Chris' PR: https://github.com/BYU-PCCL/holodeck/pull/439)

### CODE REVIEW

File: src/holodeck/environments.py [hunk 1]
```diff
@@ -470,11 +470,22 @@ def tick(self, num_ticks=1):
             self._command_center.handle_buffer()
 
             self._client.release()
-            self._client.acquire()
+            self._acquire_catch_crash()
             state = self._default_state_fn()
 
         return state
 
+    def _acquire_catch_crash(self):
```
Comment (daniekpo) [inline src/holodeck/environments.py:507]: This is great! Should be helpful with trying to pin down the semaphore busy error.

**Review by daniekpo — DISMISSED** [review]
Great job. This is very good!

**Comment by vinhowe** [comment]
@mattpow35 @chris-luangrath To test this, kill the Holodeck process while the client is running. If you use Resource Monitor on Windows, you can also go to the CPU tab, find Holodeck, right click and "suspend process" to test that the frozen error message works.

**Review by chris-luangrath — APPROVED** [review]
Seems to work correctly for me. Nice!

**Pull request merged**